tests/e2e/gc_rwx_reclaim_policy.go

```diff
@@ -0,0 +1,519 @@
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		volumeID := getVolumeIDFromSupervisorCluster(volHandle)
+		gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Creating label for PV
+		// PVC will use this label as Selector to find PV
+		staticPVLabels := make(map[string]string)
+
+		ginkgo.By("Creating the PV in guest cluster")
+		pv2 := getPersistentVolumeSpecForRWX(volHandle, v1.PersistentVolumeReclaimDelete,
+			staticPVLabels, "2Gi", "", v1.ReadWriteMany)
+		pv2, err = client.CoreV1().PersistentVolumes().Create(ctx, pv2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating the PVC in guest cluster")
+		pvc2 := getPersistentVolumeClaimSpecForRWX(namespace, staticPVLabels, pv2.Name, "2Gi")
+		pvc2, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(ctx, pvc2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Wait for PV and PVC to Bind
+		framework.ExpectNoError(fpv.WaitOnPVandPVC(client,
+			framework.NewTimeoutContextWithDefaults(), namespace, pv2, pvc2))
+
+		// Create a Pod to use this PVC and expect an error
+		ginkgo.By("Creating pod to attach PVC")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvc2}, false, execRWXCommandPod2)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod in defer function %s in namespace %s", pod.Name, pod.Namespace))
+			err = fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Delete POD
+		ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, pod.Namespace))
+		err = fpod.DeletePodWithWait(client, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Deleting the PVClaim 2")
+		framework.ExpectNoError(fpv.DeletePersistentVolumeClaim(client, pvc2.Name, pvc2.Namespace),
+			"Failed to delete PVC", pvc2.Name)
+		ginkgo.By("Sleeping for 2 min for the PVC2 to get deleted")
+		time.Sleep(2 * time.Minute)
+
+		ginkgo.By("Deleting the PV2")
+		framework.ExpectNoError(fpv.DeletePersistentVolume(client, pv2.Name))
+		ginkgo.By("Sleeping for 2 min for the PV2 to get deleted")
+		time.Sleep(2 * time.Minute)
+
+		framework.ExpectNoError(fpv.WaitForPersistentVolumeDeleted(client, pv2.Name, poll, pollTimeout))
+
+		err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
```

Comment:
This pvclaim delete is already handled in defer block line 314, We can remove this

Reply:
Fixed